### ISSUE TITLE
Trakt.tv Favorites List not being excluded

### ISSUE THREAD

**Issue by mlamoure** [rfsbraz/deleterr#27]
Hi there-

I may be configuring this wrong, but I noticed that deleterr is not picking up my Trakt “Favorites” list when I add it to my config like this:

        lists: # Trakt lists to exclude
          [
            "https://trakt.tv/movies/trending",
            "https://trakt.tv/movies/popular",
            "https://trakt.tv/movies/watched/yearly",
            "https://trakt.tv/movies/collected/yearly",
            "https://trakt.tv/users/<username>/favorites"
          ]

When running, in addition to not excluding the movies / shows in the favorites list, I see this error:

deleterr  | 2023-08-31 18:29:58 - WARNING :: trakt.py :: deleterr : Traktpy does not support watched shows. Skipping...
deleterr  | 2023-08-31 18:29:58 - WARNING :: trakt.py :: deleterr : Traktpy does not support collected shows. Skipping...

**Comment by rfsbraz** [comment]
I'll investigate thanks for the report!

**Comment by rfsbraz** [comment]
I was already aware that [TraktPY](https://traktpy.readthedocs.io/en/latest/index.html) didn't support `collected` and `watched`, hence those 2 warnings, but it seems like `favorites` is not supported as well. This libraty is missing alot of the more recent trakt features, so I may need to look for an alternative.

**Comment by mlamoure** [comment]
Maybe just document it and push this one til later?  I’ll test with a user created list.

**Comment by rfsbraz** [comment]
Yeah, I'll start investigating alternatives and add a warning about the `favorites` list in the meantime. Cheers

**Comment by deva5610** [comment]
> This libraty is missing alot of the more recent trakt features, so I may need to look for an alternative.

[pytrakt](https://pypi.org/project/pytrakt/) might suit?

**Comment by rfsbraz** [comment]
From what I can gather `pytrakt` seems to have even less support:

| Feature          | Pytrakt | TraktPy |
|------------------|---------|---------|
| Trending Movies  | ✅      | ✅      |
| Popular Movies   | ❌      | ✅      |
| Trending TV Shows| ✅      | ✅      |
| Popular TV Shows | ✅      | ✅      |
| Watchlist        | ✅      | ✅      |
| User Lists       | ✅      | ✅      |

Current solutions are either:
- Direct REST Api implementation
- Contribute to one of the projects, probably `pytrakt` because of the activity

**Comment by rfsbraz** [comment]
Thanks for your patience on this one - it's been open far too long.

To recap the root cause: your config was always correct and the URL was parsed fine, but the code path for favorites was a stub. The Trakt library deleterr uses predates Trakt's favorites API entirely, so favorites lists were silently skipped with just a log warning - the exclusion never protected anything.

#286 fixes it by calling the Trakt REST API directly for favorites (the same client id you already configured is all it needs). One caveat: your profile needs to be public, since deleterr doesn't do full OAuth. It'll be in the next release.

The `watched/yearly`-style periodic lists you also had warnings for are a separate limitation - the same direct-REST approach can fix those too, and I've noted it as follow-up work.
